### ISSUE TITLE
Indent - can not write multiline function call with indent if first arg is multiline

### ISSUE THREAD

**Issue by Alexsey** [eslint/eslint#10417]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** 10.2.1
* **npm Version:** 6.0.1

**What parser (default, Babel-ESLint, etc.) are you using?** -

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
"indent": [1, 4, {"SwitchCase": 1}]
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function someFunction () {}

// error
someFunction([
        1
    ],
    true,
    false
)

// error
someFunction({
        x: 1
    },
    true,
    false
)
```

**What did you expect to happen?**
No error

**What actually happened? Please include the actual, raw output from ESLint.**
ESLint: Expected indentation of 4 but found 8. (indent)

**Comment by platinumazure** [comment]
Hi @Alexsey, thanks for the issue.

For what it's worth, you could use `{ CallExpression: "first" }` in these cases due to how long your function names happen to be. You may wish to alter the function names so that they do not line up with 4-space indentation.

Also, I believe this style should currently be allowed by the indent rule:

```js
/* eslint indent: ["warn", 4, { SwitchCase: 1 }] */
foo(
    [
        1
    ],
    true,
    false
)

foo(
    {
        x: 1
    },
    true,
    false
)
```

**Comment by platinumazure** [comment]
@Alexsey Understood-- just trying to document what is currently available.

If you rename the function in your example to something longer than three characters, your proposal will become more clear.

Let me know if I'm reading this wrong: It looks like you want to support a double-indent after the beginning of a function call, when both a CallExpression open token (parenthesis) and ObjectExpression or ArrayExpression open token (curly or bracket, respectively) are on the same line as the function call, and the first element is on the next line. After the object/array is finished, the close token should be indented 1 level from the CallExpression, resulting in all the arguments having 1 indent level without the rule being thrown off by the open token on the same line as the CallExpression.

Have I summarized it correctly?

**Comment by platinumazure** [comment]
Edited accordingly, let me know what you think.

I agree that indent should not try to enforce object/array elements being on the same or next lines-- those are the responsibility of rules like `object-curly-newline`, `array-bracket-newline`, `object-property-newline`, and `array-element-newline`. I didn't mean to imply that indent should care about things like that.

It seems sufficient to say that if a CallExpression argument (not necessarily the first) is an object/array that goes onto the next line, then later lines should be indented at the value of the CallExpression argument indent plus the ObjectExpression or ArrayExpression indent, and the close token would be indented with just the CallExpression indent.

**Comment by Alexsey** [comment]
@platinumazure yes, this is a good rephrase
